app/src/main/java/org/dhis2/usescases/enrollment/EnrollmentActivity.kt

```diff
@@ -247,7 +247,19 @@ class EnrollmentActivity : ActivityGlobalAbstract(), EnrollmentContract.View {
     }
 
     override fun goBack() {
-        onBackPressed()
+        if (!adapter.mandatoryOk()) {
```

Comment:
can you move it to presenter and expose two methods from the view, so doing this we can unit test, cover this bug and forget forever :D